projects/pgai/pgai/vectorizer/embeddings.py

```diff
@@ -165,6 +165,26 @@ async def setup(self) -> None:  # noqa: B027 empty on purpose
         """
 
 
+class BaseURLMixin:
+    """
+    A mixin class that provides functionality for managing base URLs.
+
+    Attributes:
+        base_url (str | None): The base URL for the API.
+    """
+
+    base_url: str | None = None
+
+    def set_base_url(self, base_url: str | None):
+        """
+        Sets the base URL from the provided secrets.
+
+        Args:
+            base_url (str | None): The base URL to be set.
+        """
+        self.base_url = base_url
+
+
```

Comment:
Others might know better, but I believe it would be a bit more idiomatic to use
```
@base_url.setter
    def base_url(self, base_url):
        self._base_url = base_url
```

Reply:
Interesting. I see we are not being very consistent on the usage of annotations regarding properties. However, it seems is the way to go, as I'm seeing in other well-known libraries. 
Will make the change 💯